components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -452,93 +452,128 @@ private String getOIDCAuthzEndpoint(Map<String, String> authenticatorProperties)
+     * @param context           AuthenticationContext
+     * @param request           HttpServletRequest
+     * @param oAuthResponse     OAuthClientResponse
+     * @return The ID token
+     */
+    protected String mapIdToken(AuthenticationContext context, HttpServletRequest request, OAuthClientResponse oAuthResponse) {
+        return oAuthResponse.getParam(OIDCAuthenticatorConstants.ID_TOKEN);
+    }
 
-                jsonObject.entrySet().stream()
-                        .filter(entry -> !ArrayUtils.contains(NON_USER_ATTRIBUTES, entry.getKey()))
-                        .forEach(entry -> buildClaimMappings(claims, entry, attributeSeparator));
-                
-                authenticatedUser = AuthenticatedUser
-                        .createFederateAuthenticatedUserFromSubjectIdentifier(authenticatedUserId);
-            } else {
+    /**
+     * Retrieves or maps the access token according to the flow supported by the authenticator
+     * @param request           HttpServletRequest
+     * @param context           AuthenticationContext
+     * @param oAuthResponse     OAuthClientResponse
+     * @throws AuthenticationFailedException Throws error when access token is not found
+     */
+    protected void mapAccessToken(HttpServletRequest request, AuthenticationContext context, OAuthClientResponse oAuthResponse) throws AuthenticationFailedException {
+        String accessToken = oAuthResponse.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN);
 
-                if (log.isDebugEnabled()) {
-                    log.debug("The IdToken is null");
-                }
-                authenticatedUser = AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(
-                        getAuthenticateUser(context, jsonObject, oAuthResponse));
+        if (StringUtils.isBlank(accessToken)) {
+            if (log.isDebugEnabled()) {
+                log.debug("No access token found");
             }
+            throw new AuthenticationFailedException(ErrorMessages.ACCESS_TOKEN_EMPTY_OR_NULL.getCode(),
+                    ErrorMessages.ACCESS_TOKEN_EMPTY_OR_NULL.getMessage());
+        }
+        context.setProperty(OIDCAuthenticatorConstants.ACCESS_TOKEN, accessToken);
+    }
 
-            claims.putAll(getSubjectAttributes(oAuthResponse, authenticatorProperties));
-            authenticatedUser.setUserAttributes(claims);
+    /**
+     * Generates OAuth client and returns the oAuthResponse according to the flow supported by the authenticator
+     * @param request   HttpServletRequest
+     * @param context   AuthenticationContext
+     * @return          OAuthClientResponse
+     * @throws AuthenticationFailedException throws error when OAuthAuthzResponse validation fails for either error
+     * response or the parameters
+     */
+    protected OAuthClientResponse generateOauthResponse(HttpServletRequest request, AuthenticationContext context)
+            throws AuthenticationFailedException {
```

Comment:
Shall we fix formatting errors in this block?

Reply:
Fixed 5de101c26645ca300637dbbb70739df79837a218